### PR TITLE
cool grip shit

### PR DESCRIPTION
**Pull request by Daxz0**

### CODE REVIEW

File: src/main/actions.dsc [hunk 1]
```diff
@@ -2,8 +2,15 @@ mortal_actions:
   type: world
   debug: false
   events:
-    after player right clicks block flagged:!mortal.reviving:
+    after player left clicks block flagged:!mortal.gripping|!rclick|!mortal.reviving with:*sword|*axe:
+    - flag player lclick expire:1s
```
Comment (acikek) [inline src/main/actions.dsc]: all flags in the project should be subflags of `mortal`

File: src/main/mortal.dsc [hunk 2]
```diff
@@ -16,9 +16,10 @@ mortal:
     # Flag the player in the "dying" state
     - flag <player> mortal.dying
     - flag <entry[copy].created_npc> mortal.copy:<player>
+    #Logs player
+    - flag <player> griplog
```
Comment (acikek) [inline src/main/mortal.dsc]: unhelpful comment

File: src/main/mortal.dsc [hunk 3]
```diff
@@ -16,9 +16,10 @@ mortal:
...
     # Narrate eath message
-    - narrate "<red>You're dying! Wait for someone to revive you, or use <yellow>/mortem <red>to end your life."
-    on player dies flagged:mortal.mortem:
+    - narrate "<red>You're dying! Wait for someone to revive/grip you, or use <yellow>/mortem <red>to end your life. <&4><&l>Logging out will result in a death!"
```
Comment (acikek) [inline src/main/mortal.dsc]: logging out as a downed player who isnt being gripped shouldnt result in death. it should only be if they're being gripped

File: src/util/grave.dsc [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
     # Generate and set grave inventory
-    - define inv "<inventory[generic[size=54;title=<player.name>'s Grave]]>"
-    - inventory copy d:<[inv]> o:<player.inventory>
+    - define inv "<inventory[generic[size=54;title=<[player].if_null[<player>].name>'s Grave]]>"
+    - inventory copy d:<[inv]> o:<[player].inventory>
     - flag <[location]> mortal.grave:<[inv]>
     # Clear player's inventory
-    - inventory clear
\ No newline at end of file
+    - inventory clear d:<[player].inventory>
```
Comment (acikek) [inline src/util/grave.dsc]: why

File: src/util/grip.dsc [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    - bossbar update <[id]> progress:<[n].div[10]>
+  - flag <[target_player]> mortal.mortem
+  - run mortal_create_grave def:<[target].location>|<[target_player]>
+  - adjust <[target_player]> gamemode:survival
+  - hurt <[target_player].health> <[target_player]>
+  - flag <player> mortal.gripping:!
+  - flag <[target_player]> mortal.dying:!
+  - flag <[target_player]> griplog:!
+  - remove <[target]>
+  - bossbar remove <[id]>
```
Comment (acikek) [inline src/util/grip.dsc]: repeated code - put what you can into other tasks/procs, then modify both this and reviving code

File: src/util/grip.dsc [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  events:
+    on player quits flagged:griplog:
+    - inject mortal_find_dying_player
+    - remove <[target]>
+    - flag <player> mortal.mortem
+    - flag <player> mortal.dying:!
+    - run mortal_create_grave def:<player.location>|<player>
+    - adjust <player> gamemode:survival
+    - hurt <player.health>
+    - flag player griplog:!
```
Comment (acikek) [inline src/util/grip.dsc]: "log" should be replaced with "logout" or "quit" in all applicable cases to be more specific

**Review by acikek — CHANGES_REQUESTED** [review]

File: src/main/mortal.dsc (hunk 3 above)
Comment (Daxz0) [reply src/main/mortal.dsc]: but then... they can just leave and wait

File: src/util/grave.dsc (hunk 4 above)
Comment (Daxz0) [reply src/util/grave.dsc]: how tf do i deal with the target player?

File: src/util/grip.dsc (hunk 5 above)
Comment (Daxz0) [reply src/util/grip.dsc]: grr

File: src/util/grip.dsc (hunk 6 above)
Comment (Daxz0) [reply src/util/grip.dsc]: ok..

File: src/main/mortal.dsc (hunk 2 above)
Comment (Daxz0) [reply src/main/mortal.dsc]: well screw the comments

File: src/main/actions.dsc (hunk 1 above)
Comment (Daxz0) [reply src/main/actions.dsc]: wym

File: src/main/mortal.dsc (hunk 3 above)
Comment (acikek) [reply src/main/mortal.dsc]: so?

File: src/util/grave.dsc (hunk 4 above)
Comment (acikek) [reply src/util/grave.dsc]: i see

File: src/main/mortal.dsc (hunk 3 above)
Comment (Daxz0) [reply src/main/mortal.dsc]: oh yeah i see

File: src/util/grip.dsc (hunk 5 above)
Comment (Daxz0) [reply src/util/grip.dsc]: too hard

File: src/util/grave.dsc (hunk 4 above)
Comment (acikek) [reply src/util/grave.dsc]: you *could* also define `__player` (which `<player>` internally uses)

File: src/util/grutil.dsc [hunk 7]
```diff
@@ -0,0 +1,21 @@
+grutilbar:
```
Comment (acikek) [inline src/util/grutil.dsc]: name the file and script something actually readable. also add `mortal` in front of the script name (for example `mortal_bossbar`)

File: src/util/grutil.dsc [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+            - else if !<player.is_online> || !<[target_player].is_online>:
+                - flag <player> mortal.gripping:!
+                - flag <player> mortal.reviving:!
+                - define err "Player went offline."
+            - if <[err].exists>:
+                - bossbar update <[id]> color:red title:<red><[err]>
+                - wait 3s
+                - bossbar remove <[id]>
+                - stop
+                - bossbar update <[id]> progress:<[n].div[10]>
```
Comment (acikek) [inline src/util/grutil.dsc]: bad copypaste. pls test

File: src/util/grutil.dsc [hunk 9]
```diff
@@ -0,0 +1,21 @@
+grutilbar:
+    type: task
+    definitions: target_player|text|text2
+    script:
+        - define id grip_<[target_player].name>
+        - define title "<red><[text2]>ing <yellow><[target_player].name><green>..."
+        - bossbar create <[id]> players:<player>|<[target_player]> color:red style:segmented_10 title:<[title]> progress:0.0
+        - repeat 10 as:n:
+            - wait 10t
+            - if !<player.has_flag[mortal.gripping]>||!<player.has_flag[mortal.reviving]>:
```
Comment (acikek) [inline src/util/grutil.dsc]: spaces between operators and the things you're comparing

File: src/util/grip.dsc [hunk 10]
```diff
@@ -6,25 +6,9 @@ mortal_grip:
...
-    - else if !<player.is_online> || !<[target_player].is_online>:
-      - flag <player> mortal.gripping:!
-      - define err "Player went offline."
-    - if <[err].exists>:
-      - bossbar update <[id]> color:red title:<red><[err]>
-      - wait 3s
-      - bossbar remove <[id]>
-      - stop
-    - bossbar update <[id]> progress:<[n].div[10]>
+  - inject grutilbar defs:<[target_player]>|"Grip"|"Gripp"
```
Comment (acikek) [inline src/util/grip.dsc]: you can't use a `defs` arg for inject. you have to use the define command

Comment (acikek) [reply src/util/grip.dsc]: additionally, you don't need quotes here

**Comment by Daxz0** [comment]
when pull

**Pull request closed without merging**